### ISSUE TITLE
CET without Commit causes error

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#1842]
Not sure if that's a bug in the way we monitor or the condition for "sanity" on transactions is just too strict:

```
  | {"timestamp":"2022-04-19T01:00:51.851369243Z","level":"ERROR","fields":{"message":"Failed to rehydrate CFD: Failed to load closed CFD 408e589a-655e-4a08-9edc-82960a090741: Closed CFD has insane combination of transactions:\n                       None,\n                       None,\n                       Some(Cet { txid: Txid(754338cf0289ad7e96be8fdb9cccb3824410376040e3cd3a3a51aef764a4aee3), vout: Vout(262754), payout: Payout(Amount(0.00000000 BTC)), price: Price(40590), timestamp: Timestamp(1650074407) })
-- | --
```

**Comment by luckysori** [comment]
### Investigation

TL;DR: I cannot quite explain how we have run into this problem.

The logs reveal that this was a manually committed CFD:

```
{"message":"Appended event to database","event":"ManualCommit","order_id":"408e589a-655e-4a08-9edc-82960a090741"}
```

After the CET was confirmed sufficiently, it was moved along with other closed CFDs:


```
{"message":"Moving CFDs to closed_cfds table: [OrderId(Hyphenated(05bef017-6a09-4057-8074-5f7c42ca99c7)), OrderId(Hyphenated(408e589a-655e-4a08-9edc-82960a090741)), OrderId(Hyphenated(a5710ecb-9103-4821-9f33-a24009e212aa)), OrderId(Hyphenated(eb0365aa-e636-4ab7-aea0-febca938f101)), OrderId(Hyphenated(fa5e7ac6-ea5c-41ad-8d29-d982b5681b1b)), OrderId(Hyphenated(fb54d74b-196a-4a47-8237-bf1e42050305))]"}
```

The `event_log` table contains the `ManualCommit` event, meaning that it was part of the list of events when the CFD was being moved:

```
362|40|ContractSetupStarted|1649978140
363|40|ContractSetupCompleted|1649978145
364|40|LockConfirmed|1649978678
365|40|RolloverStarted|1649984629
366|40|RolloverAccepted|1649984629
367|40|RolloverCompleted|1649984648
368|40|LockConfirmed|1649984651
369|40|ManualCommit|1650074407
370|40|OracleAttestedPriorCetTimelock|1650074407
371|40|CommitConfirmed|1650074797
372|40|CetTimelockExpiredPostOracleAttestation|1650084251
373|40|CetConfirmed|1650087662
374|40|CetConfirmed|1650087662  
```

At the time, we had a bug which prevented us from being able to save commit transactions to the `commit_txs` table, because the table expected a `NOT NULL` column for the `vout` which the code in `db.rs` did not add (and which wasn't needed, because commit transactions only have one output). This was fixed later on with #1822.

In the logs we see that a couple of CFDs with commit transactions from the same batch failed to be moved to the `closed_cfds` table for this very reason:

```
{"message":"Failed to move closed CFD: error returned from database: NOT NULL constraint failed: commit_txs.timestamp: NOT NULL constraint failed: commit_txs.timestamp","order_id":"a5710ecb-9103-4821-9f33-a24009e212aa"}
{"message":"Failed to move closed CFD: error returned from database: NOT NULL constraint failed: commit_txs.timestamp: NOT NULL constraint failed: commit_txs.timestamp","order_id":"05bef017-6a09-4057-8074-5f7c42ca99c7"}
```

Interestingly, we don't see such an error for `408e589a-655e-4a08-9edc-82960a090741`, which makes no sense to me, because that CFD should have also failed when inserting into `commit_txs`, aborting the whole database transaction. This cannot be an instance of failing to roll back a database transaction because we clearly successfully executed code after the line of code that should have failed, since we managed to insert into the `cets` table and delete rows from `cfds` and `events`.

### Conclusion

We can safely close this issue because:

- This bug is not dangerous and it can be fixed manually by inserting into the database.
- We have already merged #1822, which solves a problem which could have had some effect on the behaviour we saw here.

**Comment by da-kami** [comment]
> At the time, we had a bug which prevented us from being able to save commit transactions to the commit_txs table, because the table expected a NOT NULL column for the vout which the code in db.rs did not add (and which wasn't needed, because commit transactions only have one output). This was fixed later on with https://github.com/itchysats/itchysats/pull/1822.

So far I can understand it, but if we failed to insert the `commit_tx` we should have failed to move the cfd into `closed_cfds`. 
⚠️ Why was the DB transaction not failing and rolled back? The CFD should have remained in `cfds` if there was a problem with moving it into `closed_cfds`! Did it actually roll back - the log indicates that it did not?
